exercise7/index.js

```diff
@@ -2,48 +2,49 @@
 window.addEventListener("load", () => {
   // Get all the elements on the page that have click actions or will be changed as the game is played.
   // For starters, I've added the message div and the reset button.
-  let rows = document.querySelectorAll(".row");
-  let listEntries = document.querySelectorAll(".listEntry");
-  let checkboxes = document.querySelectorAll("input[name=checkbox]");
+  let counter = 0;
   const submitButton = document.getElementById("submitNew");
   const addBox = document.getElementById("addNew");
   const container = document.querySelector(".container");
 
   submitButton.addEventListener("click", function () {
     let submitText = addBox.value;
 
+    const checkyes = document.querySelector("input[name=checkbox]:checked");
+
     let row = document.createElement("div");
     row.setAttribute("class", "row");
-    row.setAttribute("id", "r" + rows.length);
-    container.appendChild(row);
-    rows = document.querySelectorAll(".row");
-
+    row.setAttribute("id", "r" + counter);
+    if (checkyes) {
+      container.insertBefore(row, checkyes.parentElement.parentElement);
+    } else {
+      container.appendChild(row);
+    }
+    
     let list = document.createElement("div");
     list.setAttribute("class", "listEntry");
-    list.setAttribute("id", "l" + listEntries.length);
+    list.setAttribute("id", "l" + counter);
     list.innerHTML = submitText;
```

Comment:
This is a security risk. You shouldn't set innerHTML with any text that the user has control over. Try adding this text to your TODO list: `<img src=https://t.ly/qC7B_ onload=alert('pwned!')>`

The safe way to do this is with textContent.
```suggestion
    list.textContent = submitText;
```